cmd/flags_core.go

```diff
@@ -12,70 +12,78 @@ import (
+		coreFlag,
 		"",
-		"Indicates node to connect to the given remote core node. "+
-			"Example: <protocol>://<ip>:<port>, tcp://127.0.0.1:26657",
+		"Indicates node to connect to the given core node. "+
+			"Example: <ip>, 127.0.0.1. Assumes RPC port 26657 and gRPC port 9009 as default unless otherwise specified.",
+	)
+	flags.String(
+		coreRPCFlag,
+		"26657",
+		"Set a custom RPC port for the core node connection. The --core.ip flag must also be provided.",
 	)
 	flags.String(
 		coreGRPCFlag,
-		"",
-		"Indicates node to connect to the given celestia-app gRPC endpoint for state-related queries"+
-			"Example: <ip>:<port>, 127.0.0.1:9090",
+		"9090",
+		"Set a custom gRPC port for the core node connection. The --core.ip flag must also be provided.",
 	)
 
 	return flags
 }
 
 // ParseCoreFlags parses Core flags from the given cmd and applies values to Env.
 func ParseCoreFlags(cmd *cobra.Command, env *Env) error {
-	coreRemote := cmd.Flag(coreRemoteFlag).Value.String()
-	if coreRemote != "" {
-		proto, addr, err := validateAddress(coreRemote)
-		if err != nil {
-			return fmt.Errorf("cmd: while parsing '%s': %w", coreRemoteFlag, err)
+	coreIP := cmd.Flag(coreFlag).Value.String()
+	if coreIP == "" {
+		if cmd.Flag(coreGRPCFlag).Changed || cmd.Flag(coreRPCFlag).Changed {
+			return fmt.Errorf("cannot specify RPC/gRPC ports without specifying an IP address for --core.ip")
 		}
-
-		env.AddOptions(node.WithRemoteCore(proto, addr))
+		return nil
 	}
-
-	grpcEndpoint := cmd.Flag(coreGRPCFlag).Value.String()
-	if grpcEndpoint != "" {
-		// parse ip:port from the endpoint
-		grpcURL, err := url.Parse(grpcEndpoint)
-		if err != nil {
-			return err
-		}
-		env.AddOptions(node.WithGRPCEndpoint(grpcURL.Host))
+	// sanity check given core ip addr and strip leading protocol
+	ip, err := sanityCheckIP(coreIP)
+	if err != nil {
+		return err
 	}
 
-	return nil
-}
-
-// validateAddress parses the given address of the remote core node
-// and checks if it configures correctly
-func validateAddress(address string) (string, string, error) {
-	u, err := url.Parse(address)
+	rpc := cmd.Flag(coreRPCFlag).Value.String()
+	// sanity check rpc endpoint
+	_, err = strconv.Atoi(rpc)
 	if err != nil {
-		return "", "", err
+		return err
 	}
+	env.AddOptions(node.WithRemoteCoreIP(ip), node.WithRemoteCorePort(rpc))
 
-	if u.Scheme == "" || u.Host == "" {
-		return "", "", fmt.Errorf("both protocol and host must present in the address")
+	grpc := cmd.Flag(coreGRPCFlag).Value.String()
+	// sanity check gRPC endpoint
+	_, err = strconv.Atoi(grpc)
+	if err != nil {
+		return err
 	}
+	env.AddOptions(node.WithGRPCPort(grpc))
+	return nil
+}
 
-	if _, _, err := net.SplitHostPort(u.Host); err != nil {
-		return "", "", fmt.Errorf("incorrect address provided for Remote Core")
+// sanityCheckIP trims leading protocol scheme and port from the given
+// IP address if present.
+func sanityCheckIP(ip string) (string, error) {
+	original := ip
+	ip = strings.TrimPrefix(ip, "http://")
```

Comment:
A hit hesitant on the manual parsing of IP format, but I guess them's the breaks

Reply:
(feel free to resolve conversation at will)